examples/java/src/main/java/org/apache/beam/examples/snippets/Snippets.java

```diff
@@ -667,4 +650,126 @@ public void process(ProcessContext c) {
   }
 
   // [END AccessingValueProviderInfoAfterRunSnip1]
+
+  public static Duration gapDuration;
+
+  // [START CustomSessionWindow1]
+
+  public Collection<IntervalWindow> assignWindows(WindowFn.AssignContext c) {
+
+    // Assign each element into a window from its timestamp until gapDuration in the
+    // future.  Overlapping windows (representing elements within gapDuration of
+    // each other) will be merged.
+    return Arrays.asList(new IntervalWindow(c.timestamp(), gapDuration));
+  }
+  // [END CustomSessionWindow1]
+
+  // [START CustomSessionWindow2]
+  public static class DynamicSessions extends WindowFn<Object, IntervalWindow> {
+    /** Duration of the gaps between sessions. */
+    private final Duration gapDuration;
+
+    /** Pub/Sub attribute that modifies session gap. */
+    private final String gapAttribute;
+
+    /** Creates a {@code DynamicSessions} {@link WindowFn} with the specified gap duration. */
+    private DynamicSessions(Duration gapDuration, String gapAttribute) {
+      this.gapDuration = gapDuration;
+      this.gapAttribute = gapAttribute;
```

Comment:
In the `assignWindows` method, line [693](https://github.com/apache/beam/blob/744eed55c0a6421a3dbf7f8a7bf646f06db55c9c/examples/java/src/main/java/org/apache/beam/examples/snippets/Snippets.java#L693):

```java
dataDrivenGap = Duration.standardSeconds(message.gapAttribute);
```

Reply:
Indeed it looks like you are correct. In my example I used `JSONObject` instead of `Pojo` so I did not make these changes. In its current implementation I am not sure it would work when using [`withGapAttribute()`](https://github.com/apache/beam/blob/dcf0270ae9a253057a44a21cb982728d53cb05db/examples/java/src/main/java/org/apache/beam/examples/snippets/Snippets.java#L796)